### ISSUE TITLE
Increase Coverage of Mutli Node Consolidation suite_tests

### ISSUE THREAD

**Issue by njtran** [aws/karpenter-core#324]
### Tell us about your request

Multi node consolidation is a core deprovisioner in the deprovisioning controller, but only has 3 suite tests. I also made changes to the core logic and `make presubmit` didn't notice the change was made, which we should be catching.

### Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?

--

### Are you currently working around this issue?

---

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by khareyash05** [comment]
I would like to add unit tests for the functions

**Comment by khareyash05** [comment]
@jonathan-innis which files I need to work on except `multimachineconsolidation.go`

**Comment by khareyash05** [comment]
we are  talking about `singlemachineconsolidation.go` and `multimachineconsolidation.go` right?

**Comment by jonathan-innis** [comment]
Yes, the tests for these files lives inside of `suite_test.go` right now. There's handling for multi-node consolidation [here](https://github.com/aws/karpenter-core/blob/main/pkg/controllers/deprovisioning/suite_test.go#L2228) and single-node consolidation is [all over the file](https://github.com/aws/karpenter-core/blob/main/pkg/controllers/deprovisioning/suite_test.go#L218)

**Comment by Bryce-Soghigian** [comment]
@khareyash05  are you still working on this?

**Comment by khareyash05** [comment]
> @khareyash05  are you still working on this? 

Please go ahead with this

**Comment by jonathan-innis** [comment]
/unassign

I'm going to unassign since it looks like there's been no progress on this since it was assigned.
